src/middlewares/auth-middleware.ts

```diff
@@ -41,150 +41,151 @@ async function authentication(
       cookieSalt: config.cookieSalt,
     });
 
-    if (userCookie) {
-      span.setDisclosedAttribute("user cookie found", true);
-      const { access_token: currentAccessToken, sub } = userCookie;
-
-      await decryptVerifyAccessToken(currentAccessToken).catch(
-        async (error) => {
-          if (error.name === "TokenExpiredError") {
-            span.setDisclosedAttribute("is access_token expired", true);
-            const user = await getDBUserFromSub(sub).catch((error) => {
-              span.setDisclosedAttribute("database reachable", false);
-              throw webErrorFactory({
-                ...webErrors.databaseUnreachable,
-                parentError: error,
-              });
-            });
+    if (!userCookie) {
+      span.setDisclosedAttribute("user cookie found", false);
+      response.statusCode = HttpStatus.TEMPORARY_REDIRECT;
+      response.setHeader("location", "/");
+      return;
+    }
+    span.setDisclosedAttribute("user cookie found", true);
+
+    const { access_token: currentAccessToken, sub } = userCookie;
+    const user = await getDBUserFromSub(sub).catch((error) => {
+      span.setDisclosedAttribute("database reachable", false);
+      throw webErrorFactory({
+        ...webErrors.databaseUnreachable,
+        parentError: error,
+      });
+    });
 
-            span.setDisclosedAttribute("database reachable", true);
-
-            if (user) {
-              span.setDisclosedAttribute("user found on DB", true);
-              const {
-                refresh_token,
-                access_token,
-              } = await oauth2Client
-                .refreshTokens(user.refresh_token)
-                .catch((error) => {
-                  span.setDisclosedAttribute("is token refreshed", false);
-                  if (error instanceof UnreachableError) {
-                    throw webErrorFactory({
-                      ...webErrors.unreachable,
-                      parentError: error,
-                    });
-                  }
-
-                  throw error;
+    if (!user) {
+      span.setDisclosedAttribute("db user found", false);
+      response.statusCode = HttpStatus.TEMPORARY_REDIRECT;
+      response.setHeader("location", "/");
+      return;
+    }
+    span.setDisclosedAttribute("db user found", true);
```

Comment:
Does this part is useful right here ? Since there's still a check on user inside `decryptVerifyAccessToken` and a `else` case which does the same thing (line 155-160) ? 🤔